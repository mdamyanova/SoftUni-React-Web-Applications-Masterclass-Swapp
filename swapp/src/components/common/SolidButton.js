import React from 'react';
import PropTypes from 'prop-types';
import { Button } from 'rebass';

const SolidButton = ({ text, onClick, style }) => {
  return (
    <Button
      fontWeight='bold'
      onClick={onClick}
      css={`
        background-color: var(--button-background-color);
        color: var(--button-font-color);
      `}
      sx={{ ...style }}
    >
      {text}
    </Button>
  );
};

SolidButton.propTypes = {
  onClick: PropTypes.func.isRequired,
  text: PropTypes.string.isRequired,
  style: PropTypes.object
};

SolidButton.defaultProps = {
  style: {}
};
export default SolidButton;
