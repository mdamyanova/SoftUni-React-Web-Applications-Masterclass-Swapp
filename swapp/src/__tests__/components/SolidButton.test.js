import React from "react";
import { shallow, mount } from "enzyme";
import SolidButton from "../../components/common/forms/SolidButton";

describe("<SolidButton />", () => {
  it("should match snapshot", () => {
    expect(
      shallow(<SolidButton onClick={() => {}} text="Button Text" />)
    ).toMatchSnapshot();
  });

  it("should have props", () => {
    const wrapper = mount(
      <SolidButton onClick={() => {}} text="Button Text" />
    );
    expect(wrapper.props().text).toEqual("Button Text");
  });
});
