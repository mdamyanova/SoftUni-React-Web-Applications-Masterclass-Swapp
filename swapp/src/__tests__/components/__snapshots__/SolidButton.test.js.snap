// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SolidButton /> should match snapshot 1`] = `
<ForwardRef
  css="
        background-color: var(--button-background-color);
        color: var(--button-font-color);
      "
  fontWeight="bold"
  onClick={[Function]}
  sx={Object {}}
>
  Button Text
</ForwardRef>
`;
